services/src/main/java/org/keycloak/services/managers/ApplianceBootstrap.java

```diff
@@ -106,17 +114,23 @@ public boolean createMasterRealm() {
         return true;
     }
 
-    public void createMasterRealmUser(String username, String password) {
+    public void createTemporaryMasterRealmAdminUser(String username, String password, Integer expriationMinutes, boolean initialUser) {
```

Comment:
Just one last nitpick (sorry, missed it before). I'd remove the `expirationMinutes` arg for now as it might be confusing to have something in the method signature that doesn't work.

Reply:
@vmuzikar my assumption was that we didn't care about that given that the method name is misleading as well. Do you want me to change the method names as well?

Reply:
I believe the name is correct, the admin should still be considered temporary even though it's not expired.

Reply:
Okay, we'll consider it temporary but non-expiring :)